app/controllers/api_controller/parser/request.rb

```diff
@@ -0,0 +1,81 @@
+class ApiController
+  module Parser
+    class Request
```

Comment:
WDYT about calling this a `RequestAdapter` or similar, since it is largely changing/augmenting `ActionDispatch::Request`?